django/db/backends/base/schema.py

```diff
@@ -612,7 +623,12 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
                     self.execute(self._delete_index_sql(model, index_name))
         # Change check constraints?
         if old_db_params['check'] != new_db_params['check'] and old_db_params['check']:
-            constraint_names = self._constraint_names(model, [old_field.column], check=True)
+            meta_constraint_names = {constraint.name for constraint in model._meta.constraints}
+            constraint_names = [
+                name
+                for name in self._constraint_names(model, [old_field.column], check=True)
+                if name not in meta_constraint_names
+            ]
```

Comment:
I wonder :thinking: if we could move this repetitive logic directly into `_constraint_names()`, e.g.
```diff
diff --git a/django/db/backends/base/schema.py b/django/db/backends/base/schema.py
index 0522abcaad..daeb210ad7 100644
--- a/django/db/backends/base/schema.py
+++ b/django/db/backends/base/schema.py
@@ -1116,15 +1116,17 @@ class BaseDatabaseSchemaEditor:
         with self.connection.cursor() as cursor:
             constraints = self.connection.introspection.get_constraints(cursor, model._meta.db_table)
         result = []
+        meta_constraint_names = {constraint.name for constraint in model._meta.constraints}
+        meta_index_names = {constraint.name for constraint in model._meta.indexes}
         for name, infodict in constraints.items():
             if column_names is None or column_names == infodict['columns']:
-                if unique is not None and infodict['unique'] != unique:
+                if unique is not None and (infodict['unique'] != unique or name in meta_constraint_names):
                     continue
                 if primary_key is not None and infodict['primary_key'] != primary_key:
                     continue
-                if index is not None and infodict['index'] != index:
+                if index is not None and (infodict['index'] != index or name in meta_index_names):
                     continue
-                if check is not None and infodict['check'] != check:
+                if check is not None and (infodict['check'] != check or name in meta_constraint_names):
                     continue
                 if foreign_key is not None and not infodict['foreign_key']:
                     continue
```

Reply:
moved